### PR TITLE
Dockerize the app

### PR DESCRIPTION
**Pull request by fredyw**
Enterprise software keeps on evolving, it's time to upgrade our ancient
build system to the latest and greatest as of this writing (4.1). Lastly
enterprise software should definitely be containerized.

### CODE REVIEW

**Comment by GabLeRoux** [comment]
related to #335 :trollface:

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+ENV APP_HOME=/fizzbuzz \
+    JAR_FILE=FizzBuzzEnterpriseEdition-all.jar
+RUN mkdir -p ${APP_HOME}
+
+COPY build/libs/${JAR_FILE} ${APP_HOME}
+
+WORKDIR ${APP_HOME}
+
+CMD exec java -jar ${JAR_FILE}
```
Comment (taylorwood) [inline Dockerfile]: This should probably also use the new experimental JVM options for cgroup memory limits:

```-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap```

https://bugs.openjdk.java.net/browse/JDK-8170888

Comment (fredyw) [reply Dockerfile]: Done!

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM openjdk:8-jre
```
Comment (anescobar1991) [inline Dockerfile]: Should we build our own base image instead?

File: dockerize.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+./gradlew
+docker build -t fizzbuzz-enterprise-edition .
+docker run -it --rm --name fizzbuzz-enterprise-edition fizzbuzz-enterprise-edition
```
Comment (eosrei) [inline dockerize.sh]: We require the use of long form arguments for documentation purposes and all dockerized applications container images must be clearly namespaced. Please correct these discrepancies before this git pull request can be merged.
```
docker build \
  --rm \
  --tag enterprisequalitycoding/fizzbuzz-enterprise-edition \
  --file Dockerfile \
  .

docker run  \
  --interactive \
  --tty \
  --rm \
  --name fizzbuzz-enterprise-edition \
  enterprisequalitycoding/fizzbuzz-enterprise-edition
```

File: Dockerfile (hunk 2 above)
Comment (agilob) [reply Dockerfile]: My company just started migration from Java 6 to Java 7, our code works and tests pass on 7 but we won't be deploying Java7 any time soon on production. Could you please change it to Java 6 and use Oracle Java, so more companies worldwide can benefit from your contribution?

**Comment by rchargel** [comment]
I really need this docker PR to go in so I can set this up in Kubernetes for my Cloud-Scale deployment.  Also, will this work with a MangoDB backend; it's web-scale.

**Comment by emiln** [comment]
As a company we are deeply grateful for the many open source contributions we're receiving, but [we can no longer](https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition/issues/472) in good faith support merge requests into the problematically named `master` branch, which has been removed.

Instead we hope that you will reopen your merge request and target the new default branch, artisanally crafted by our Swedish Inclusivity Consultant, called `uinverse`. You can read more about the inclusivity success story [in the issue](https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition/issues/472).

**Pull request closed without merging**